### PR TITLE
[makeid] Add new port

### PR DESCRIPTION
**Pull request by Pospelove**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/makeid/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,17 @@
+vcpkg_download_distfile(ARCHIVE
+    URLS "http://www.humus.name/3D/MakeID.h"
+    FILENAME "MakeID.h"
+    SHA512 9b7cb5c1b71904f37f65fcac3d18194154029fbe04d89099d879ce8eb03e796662c78653322317ed72988d3695414aaa6e6c24cfff999bea5009ec47119c57a7
+)
+
+file(COPY ${ARCHIVE} DESTINATION ${CURRENT_PACKAGES_DIR}/include)
```
Comment (Adela0814) [inline ports/makeid/portfile.cmake:7]:
```suggestion
file(COPY ${ARCHIVE} DESTINATION "${CURRENT_PACKAGES_DIR}/include")
```

File: ports/makeid/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+set(license_text 
+"Public Domain
+
+This file is released in the hopes that it will be useful. Use in whatever way you like, but no guarantees that it
+actually works or fits any particular purpose. It has been unit-tested and benchmarked though, and seems to do
+what it was designed to do, and seems pretty quick at it too."
+)
+
+file(WRITE ${CURRENT_PACKAGES_DIR}/share/makeid/copyright "${license_text}")
```
Comment (Adela0814) [inline ports/makeid/portfile.cmake:17]:
```suggestion
file(WRITE "${CURRENT_PACKAGES_DIR}/share/makeid/copyright" "${license_text}")
```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Review by Adela0814 — DISMISSED** [review]

File: ports/makeid/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,17 @@
+vcpkg_download_distfile(ARCHIVE
+    URLS "http://www.humus.name/3D/MakeID.h"
+    FILENAME "MakeID.h"
```
Comment (dg0yt) [inline ports/makeid/portfile.cmake:3]: There can't be different versions (and SHA512) with the same name.
Can we use `FILENNAME "MakeID.h-${VERSION}"` here (and in the copy command below)?

**Comment by BillyONeal** [comment]
I'm a bit concerned over 'makeid' since that seems likely to be a conflicting name but I can't find any actual conflicts so I guess this works.

If you know how to contract upstream it would be good to ask them to make versioned download locations.

**Comment by dg0yt** [comment]
> I'm a bit concerned over 'makeid' since that seems likely to be a conflicting name.

In that case we should quickly move it to a subdir...

**Comment by BillyONeal** [comment]
> > I'm a bit concerned over 'makeid' since that seems likely to be a conflicting name.
> 
> In that case we should quickly move it to a subdir...

I didn't find any actual conflicts so I don't think it needs a subdir. Unless we see actual conflicts.

**Pull request merged**